File: tasks/2022/qualification/goose.cpp
/*
Minden kockara szimulaljuk a jatekot. 
Az i-edik kockaval dobva egy kor soran eloszor i-t lepunk, majd addig ugralunk elore, amig az alabbi ket feltetel mindegyike teljesul:
1. Nem ertunk a palya vegere.
2. Az ugrani kivant hossz pozitiv (azaz nem 0).
*/

#include <bits/stdc++.h>
using namespace std;
const int N = 1e5;
int n, j[N];

int main()
{
    cin >> n;
    for (int i = 0; i < n; ++i) cin >> j[i];
    for (int i = 1; i <= 6; ++i)
    {
        int p = 0, a = 0;
        while (p < n)
        {
            p += i, a++;
            while (p < n && j[p] != 0) p += j[p];
        }
        cout << a << " ";
    }
}

File: tasks/2022/qualification/goose.py
'''
Minden kockara szimulaljuk a jatekot. 
Az i-edik kockaval dobva egy kor soran eloszor i-t lepunk, majd addig ugralunk elore, amig az alabbi ket feltetel mindegyike teljesul:
1. Nem ertunk a palya vegere.
2. Az ugrani kivant hossz pozitiv (azaz nem 0).
'''

n = int(input())
j = list(map(int, input().split()))

for i in range(1, 7):
    p, a = 0, 0
    while p < n:
        p += i
        a += 1
        while p < n and j[p] != 0:
            p += j[p]
    print(a, end = ' ')